connector-builder-server/src/main/openapi/openapi.yaml

```diff
@@ -0,0 +1,246 @@
+              sliceId:
+                type: string
+                description: String identifying the current slice
+              pages:
+                type: array
+                description: The pages returned from the read command
+                items:
+                  type: object
+                  required:
+                    - pageId
+                    - results
+                    - logs
+                    - request
+                    - response
+                  properties:
+                    pageId:
+                      type: string
+                      description: String identifying the current page
+                    results:
+                      type: array
+                      description: The RECORD and STATE AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    logs:
+                      type: array
+                      description: The LOG AirbyteMessages coming from the read operation for this page
+                      items:
+                        type: string
+                    request:
+                      type: object
+                      description: The HTTP request sent to the source API for this page
+                      required:
+                        - url
+                        - headers
+                      properties:
+                        url:
+                          type: string
+                          description: URL that the request was sent to
+                        body:
+                          type: object
+                          description: The body of the HTTP request, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP request
+                    response:
+                      type: object
+                      description: The response from the source API for this page
+                      required:
+                        - status
+                        - headers
+                      properties:
+                        status:
+                          type: integer
+                          description: The status of the response
+                        body:
+                          type: object
+                          description: The body of the HTTP response, if present
+                        headers:
+                          type: object
+                          description: The headers of the HTTP response
+    StreamReadRequestBody:
+      type: object
+      required:
+        - definition
+        - stream
+      properties:
+        definition:
+          $ref: "#/components/schemas/ConnectorDefinitionBody"
+          description: The config-based connector definition contents
+        stream:
+          type: string
+          description: Name of the stream to read
+        numSlices:
```

Comment:
open question: do we want to request a number of slices or specify which slice to request?

Reply:
What would it look like to specify a specific slice to request here? Passing in a state blob, or something else?

Also, based on our discussion this morning around the `Next page` / `Next slice` behavior this morning, I think we'll want to update this part of the API to accommodate that behavior